server/challenge.ts

```diff
@@ -0,0 +1,246 @@
+  account: string;
+  home_domain: string;
+  client_domain: string | undefined;
+};
+
+export type ChallengeResponse = {
+  authorization_entries: string;
+  network_passphrase: string;
+};
+
+export async function getChallenge(
+  request: ChallengeRequest,
+): Promise<ChallengeResponse> {
+  const sourceAccount = await rpc.getAccount(sourceKeypair.publicKey());
+
+  let clientDomainAddress: string | undefined = undefined;
+  if (request.client_domain !== undefined) {
+    clientDomainAddress = await fetchSigningKey(request.client_domain);
+  }
+  const nonce = await generateNonce(request.account);
+
+  const fields = [
+    new xdr.ScMapEntry({
+      key: xdr.ScVal.scvSymbol("account"),
+      val: nativeToScVal(request.account),
+    }),
+    ...(request.client_domain !== undefined
+      ? [
+        new xdr.ScMapEntry({
+          key: xdr.ScVal.scvSymbol("client_domain"),
+          val: nativeToScVal(request.client_domain),
+        }),
+        new xdr.ScMapEntry({
+          key: xdr.ScVal.scvSymbol("client_domain_account"),
+          val: nativeToScVal(clientDomainAddress),
+        }),
+      ]
+      : []),
+    new xdr.ScMapEntry({
+      key: xdr.ScVal.scvSymbol("home_domain"),
+      val: nativeToScVal(request.home_domain),
+    }),
+    new xdr.ScMapEntry({
+      key: xdr.ScVal.scvSymbol("nonce"),
+      val: nativeToScVal(nonce),
+    }),
+    new xdr.ScMapEntry({
+      key: xdr.ScVal.scvSymbol("web_auth_domain"),
+      val: nativeToScVal(request.home_domain),
+    }),
+    new xdr.ScMapEntry({
+      key: xdr.ScVal.scvSymbol("web_auth_domain_account"),
+      val: nativeToScVal(sep10SigningKeypair.publicKey()),
+    }),
+
+  ];
+
+  const args = [
+    xdr.ScVal.scvMap(fields),
+  ];
+  const builtTransaction = new TransactionBuilder(sourceAccount, {
+    fee: BASE_FEE,
+    networkPassphrase: Networks.TESTNET,
+  })
+    .addOperation(webAuthContract.call("web_auth_verify", ...args))
+    .setTimeout(300)
+    .build();
+
+  // Simulate the transaction to get the authorization entries
+  const simulatedTransaction = await rpc.simulateTransaction(builtTransaction);
+  // Check if the response is a success
+  let authEntries: xdr.SorobanAuthorizationEntry[];
+  if ("result" in simulatedTransaction) {
+    const result = simulatedTransaction.result!;
+    authEntries = result.auth;
+  } else {
+    throw new Error("Transaction simulation failed");
+  }
+
+  // Sign the server's authorization entry
+  const finalAuthEntries = authEntries.map(async (entry) => {
+    if (
+      entry.credentials().switch() ===
+        xdr.SorobanCredentialsType.sorobanCredentialsAddress() &&
+      entry.credentials().address().address().switch() ===
+        xdr.ScAddressType.scAddressTypeAccount()
+    ) {
+      const validUntilLedgerSeq = (await rpc.getLatestLedger()).sequence + 1;
+      const signed = await authorizeEntry(
+        entry,
+        sep10SigningKeypair,
+        validUntilLedgerSeq,
+        Networks.TESTNET,
```

Comment:
There are a few instances where the network is hardcoded with `Networks.TESTNET`.

Can we store it on the envs and default to TESTNET when the env is not set? Or at least have a centralized constant that can be reused?